lib/scholar/cross_decomposition/pls_svd.ex

```diff
@@ -0,0 +1,353 @@
+
+  ## Options
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Return Values
+
+  Returns tuple with transformed data `{x_transformed, y_transformed}` where:
+
+    * `x_transformed` is `{num_samples, num_features}` shaped tensor.
+    
+    * `y_transformed` is `{num_samples, num_features}` shaped tensor.
+
+  ## Examples
+
+      iex> x = Nx.tensor([[0.0, 0.0, 1.0],
+      ...>                [1.0, 0.0, 0.0],
+      ...>                [2.0, 2.0, 2.0],
+      ...>                [2.0, 5.0, 4.0]])
+      iex> y = Nx.tensor([[0.1, -0.2],
+      ...>                [0.9, 1.1],
+      ...>                [6.2, 5.9],
+      ...>                [11.9, 12.3]])
+      iex> {x, y} = Scholar.CrossDecomposition.PLSSVD.fit_transform(x, y)
+      iex> x
+      #Nx.Tensor<
+        f32[4][2]
+        [
+          [-1.397004246711731, -0.10283949971199036],
+          [-1.1967883110046387, 0.17159013450145721],
+          [0.5603229403495789, -0.10849219560623169],
+          [2.0334696769714355, 0.039741579443216324]
+        ]
+      >
+      iex> y
+      #Nx.Tensor<
+        f32[4][2]
+        [
+          [-1.2260178327560425, -0.019306711852550507],
+          [-0.9602956175804138, 0.04015407711267471],
+          [0.3249155580997467, -0.04311027377843857],
+          [1.8613981008529663, 0.022262824699282646]
+        ]
+      >
+
+  """
+
+  deftransform fit_transform(x, y, opts \\ []) do
+    fit_transform_n(x, y, NimbleOptions.validate!(opts, @opts_schema))
+  end
+
+  defnp fit_transform_n(x, y, opts) do
+    fit(x, y, opts)
+    |> transform(x, y, opts)
+  end
+
+  defnp check_x_y(x, y, opts) do
+    y =
+      case Nx.shape(y) do
+        {n} -> Nx.reshape(y, {n, 1})
+        _ -> y
+      end
+
+    num_components = opts[:num_components]
+    {num_samples, num_features} = Nx.shape(x)
+    {num_samples_y, num_targets} = Nx.shape(y)
+
+    cond do
+      num_samples != num_samples_y ->
+        raise ArgumentError,
+              """
+              num_samples must be the same for x and y \
+              x num_samples = #{num_samples}, y num_samples = #{num_samples_y}
+              """
+
+      num_components > num_features ->
+        raise ArgumentError,
+              """
+              num_components must be less than or equal to \
+              num_features = #{num_features}, got #{num_components}
+              """
+
+      num_components > num_samples ->
+        raise ArgumentError,
+              """
+              num_components must be less than or equal to \
+              num_samples = #{num_samples}, got #{num_components}
+              """
+
+      num_components > num_targets ->
+        raise ArgumentError,
+              """
+              num_components must be less than or equal to \
+              num_targets = #{num_targets}, got #{num_components}
+              """
+
+      true ->
+        nil
+    end
+
+    {x, y}
+  end
+
+  defnp center_scale_x_y(x, y, opts) do
```

Comment:
In this function, all the ops are the same for both x and y. I think it will be better to write this to a single tensor and apply function twice